File: 1620-check-if-array-pairs-are-divisible-by-k/1620-check-if-array-pairs-are-divisible-by-k.cs
public class Solution {
    public bool CanArrange(int[] arr, int k) {
        
        Dictionary<int,int> dic = new Dictionary<int,int>();
        for(int i = 0; i< arr.Length; i++)
        {
            int rem = ((arr[i]%k)+k)%k;
            if(dic.ContainsKey(rem))
                dic[rem]++;
            else
                dic.Add(rem,1);
        }
        foreach(var x in dic)
        {
            if(x.Key ==0 && x.Value %2 !=0)
                return false;
            if(x.Key !=0 && (!dic.ContainsKey(k-x.Key) || dic[k-x.Key] != x.Value))
                return false;
        }return true;
        
    }
}

File: 1620-check-if-array-pairs-are-divisible-by-k/1620-check-if-array-pairs-are-divisible-by-k.cpp
class Solution {
public:
    bool canArrange(vector<int>& arr, int k) {

        unordered_map<int,int> mp;
        for(int i = 0; i< arr.size(); i++)
        {
            int rem = ((arr[i]%k)+k)%k;
            mp[rem]++;
        }
        for(auto x: mp)
        {
            if(x.first ==0 && x.second %2 !=0)
                return false;
            if(x.first !=0 && mp[k-x.first] != x.second)
                return false;
        }return true;
        
    }
};